addons/udes_stock/models/stock_picking.py

```diff
@@ -214,6 +215,8 @@ def create(self, vals):
         picking = super().create(vals)
         if not picking.backorder_id:
             picking.write({"u_original_picking_id": picking.id})
+        # Just in case date_done field is set to datetime.now but picking is not done
+        picking.safe_prevent_date_done_on_not_done_pickings()
```

Comment:
nit: maybe should be called something slightly different, like
```python
picking.clear_date_done_if_not_done()
```

Reply:
We could simplify it further, e.g. `check_date_done`.